app/assets/javascripts/Components/markdown_preview.jsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+export default class MarkdownPreview extends React.Component {
+  componentDidMount = () => {
+    const target_id = "annotation_preview";
+    MathJax.Hub.Queue(["Typeset", MathJax.Hub, target_id]);
+  };
+
+  componentWillUnmount = () => {
```

Comment:
I think I know what you're trying to do here, but it's not the right way to solve the problem. You can pass the `forceRender` prop to the "write" tab so that it's always rendered, to ensure that the event listeners for the autocomplete stay. (That's probably the simplest approach. Another option would have been to make the text input its own react component, and have the event listeners registered within that component.)